### PR TITLE
test(iam): Add remaining GCP tests for IAM checks

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

There was some checks without tests for IAM in GCP provider.

### Description

Added new tests for:

- [x] iam_sa_no_administrative_privileges check
- [x] iam_sa_no_user_managed_keys check
- [x] iam_sa_user_managed_key_rotate_90_days check

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/providers/gcp/services/iam/iam_sa_no_administrative_privileges/iam_sa_no_administrative_privileges_test.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
+from unittest import mock
+
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    GCP_US_CENTER1_LOCATION,
+    set_mocked_gcp_provider,
+)
+
```
Comment (puchy22) [inline tests/providers/gcp/services/iam/iam_sa_no_administrative_privileges/iam_sa_no_administrative_privileges_test.py:8]: Please add a tests case with no service accounts and when the binding and the service account email do not match.

File: tests/providers/gcp/services/iam/iam_sa_no_user_managed_keys/iam_sa_no_user_managed_keys_test.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    GCP_US_CENTER1_LOCATION,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_iam_sa_no_user_managed_keys:
+    def test_iam_sa_no_keys(self):
```
Comment (puchy22) [inline tests/providers/gcp/services/iam/iam_sa_no_user_managed_keys/iam_sa_no_user_managed_keys_test.py:34]: Please add a test case for no service accounts

File: tests/providers/gcp/services/iam/iam_sa_user_managed_key_rotate_90_days/iam_sa_user_managed_key_rotate_90_days_test.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                ServiceAccount(
+                    name="projects/my-project/serviceAccounts/my-service-account@my-project.iam.gserviceaccount.com",
+                    email="my-service-account@my-project.iam.gserviceaccount.com",
+                    display_name="My service account",
+                    keys=[
+                        Key(
+                            name="projects/my-project/serviceAccounts/my-service-account@my-project.iam.gserviceaccount.com/keys/90c48f61c65cd56224a12ab18e6ee9ca9c3aee7c",
+                            origin="GOOGLE_PROVIDED",
+                            type="USER_MANAGED",
+                            valid_after=datetime.strptime("2024-07-10", "%Y-%m-%d"),
```
Comment (puchy22) [inline tests/providers/gcp/services/iam/iam_sa_user_managed_key_rotate_90_days/iam_sa_user_managed_key_rotate_90_days_test.py]: Please add other date or mock the `datetime.now()` value to ensure the test is not going to expirate

**Review by puchy22 — CHANGES_REQUESTED** [review]
Please review my comments.

**Comment by MrCloudSec** [comment]
@danibarranqueroo please, update the PR description. Thanks!

File: tests/providers/gcp/services/iam/iam_sa_no_administrative_privileges/iam_sa_no_administrative_privileges_test.py (hunk 1 above)
Comment (danibarranqueroo) [reply tests/providers/gcp/services/iam/iam_sa_no_administrative_privileges/iam_sa_no_administrative_privileges_test.py:8]: Done.

File: tests/providers/gcp/services/iam/iam_sa_no_user_managed_keys/iam_sa_no_user_managed_keys_test.py (hunk 2 above)
Comment (danibarranqueroo) [reply tests/providers/gcp/services/iam/iam_sa_no_user_managed_keys/iam_sa_no_user_managed_keys_test.py:34]: Done.

File: tests/providers/gcp/services/iam/iam_sa_user_managed_key_rotate_90_days/iam_sa_user_managed_key_rotate_90_days_test.py (hunk 3 above)
Comment (danibarranqueroo) [reply tests/providers/gcp/services/iam/iam_sa_user_managed_key_rotate_90_days/iam_sa_user_managed_key_rotate_90_days_test.py]: Done.

**Review by puchy22 — APPROVED** [review]
Nice job 💯

**Review by pedrooot — APPROVED** [review]
LGTM! Thanks @danibarranqueroo

**Pull request merged**